components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go

```diff
@@ -32,13 +50,230 @@ func (ss *BatchCheckService) BatchCheck(checks []string, config models.Config) m
 		checkTriggerRespMap = getBastionCheckResp(ss, bastionChecks, config)
 	}
 
-	// Get remote check trigger resp and append to checkTriggerRespMap (keys and values)
+	successfullyStartedMockServers, notStartedMockServers, err := ss.startMockServerIfNeeded(remoteChecks, config)
+	if err != nil {
+		ss.log.Error("Error occurred while starting mock server")
+		return models.BatchCheckResponse{}, err
+	}
+
+	if len(notStartedMockServers) > 0 {
+		for _, successfullyStartedMockServer := range successfullyStartedMockServers {
+			if successfullyStartedMockServer.Host != "" {
+				go ss.StopMockServerOnHostAndPort(successfullyStartedMockServer.Host, successfullyStartedMockServer.Protocol, successfullyStartedMockServer.Port)
+			}
+		}
+		return models.BatchCheckResponse{}, errors.New("mock server not started successfull on some nodes")
+	}
 	if len(remoteChecks) > 0 {
 		for key, value := range getRemoteCheckResp(ss, remoteChecks, config) {
```

Comment:
Please keep it is simple to read, one line doing one thing and not multiple things. Call the function and save the results and then iterate over the results - @rishabhjhs 

Also, this function `getRemoteCheckResp` actually is running remote checks, why not just name it that: `runRemoteChecks`
Also the function itself is asking for BatchCheckService instance, and it is present in this same file, why not make it a method of this service ? @daveaugustus  @rishabhjhs  @Dmaddu